### PR TITLE
Remove code that is unused in .NET 8

### PR DESCRIPTION
**Pull request by MichelZ**
There's code marked as !NET8_0_OR_GREATER in the netcore project, or NET && !NET8_0_OR_GREATER in the shared project that can be removed, now that the only supported .NET (Core) version is 8+

### CODE REVIEW

**Comment by ErikEJ** [comment]
image

"I love the smell of dead code in the morning" 😄

**Comment by MichelZ** [comment]
Should also fix #303 I think

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj [hunk 1]
```diff
@@ -217,9 +217,6 @@
     <Compile Include="$(CommonSourceRoot)Microsoft\Data\ProviderBase\TimeoutTimer.cs">
       <Link>Microsoft\Data\ProviderBase\TimeoutTimer.cs</Link>
     </Compile>
-    <Compile Include="$(CommonSourceRoot)Microsoft\Data\SqlClient\SSPI\ManagedSSPIContextProvider.cs">
```
Comment (ErikEJ) [inline src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj:220]: Were these files not used in .NET Framework?

File: src/Microsoft.Data.SqlClient/src/Interop/Windows/Interop.Libraries.cs [hunk 2]
```diff
@@ -9,9 +9,8 @@ internal static partial class Libraries
         internal const string Kernel32 = "kernel32.dll";
         internal const string NtDll = "ntdll.dll";
 #if !NET8_0_OR_GREATER
```
Comment (ErikEJ) [inline src/Microsoft.Data.SqlClient/src/Interop/Windows/Interop.Libraries.cs:11]: Why was this left in?

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj (hunk 1 above)
Comment (MichelZ) [reply src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj:220]: That is the #IF NET6_0 file...   so why it was referenced, it was not compiled for netfx

File: src/Microsoft.Data.SqlClient/src/Interop/Windows/Interop.Libraries.cs (hunk 2 above)
Comment (MichelZ) [reply src/Microsoft.Data.SqlClient/src/Interop/Windows/Interop.Libraries.cs:11]: The tests use that in multiple locations in SecurityWrapper.cs
https://github.com/dotnet/SqlClient/blob/4a9b070fee2a4267ef48a7052efa67af2eaf8ea5/src/Microsoft.Data.SqlClient/tests/tools/TDS/TDS.EndPoint/SSPI/SecurityWrapper.cs#L27

**Review by benrr101 — APPROVED** [review]
Hot dang! I love these kinds of changes! My only complaint is you stole all the fun from me :P

**Review by cheenamalhotra — APPROVED** [review]
Thanks @MichelZ 
LGTM!

**Pull request merged**